templates/custom/dg_paragraphs/image-gallery/field--paragraph--field-image--image.html.twig

```diff
@@ -0,0 +1,90 @@
+{#
+/**
+ * @file
+ * Theme override for a field.
+ *
+ * To override output, copy the "field.html.twig" from the templates directory
+ * to your theme's directory and customize it, just like customizing other
+ * Drupal templates such as page.html.twig or node.html.twig.
+ *
+ * Instead of overriding the theming for all fields, you can also just override
+ * theming for a subset of fields using
+ * @link themeable Theme hook suggestions. @endlink For example,
+ * here are some theme hook suggestions that can be used for a field_foo field
+ * on an article node type:
+ * - field--node--field-foo--article.html.twig
+ * - field--node--field-foo.html.twig
+ * - field--node--article.html.twig
+ * - field--field-foo.html.twig
+ * - field--text-with-summary.html.twig
+ * - field.html.twig
+ *
+ * Available variables:
+ * - attributes: HTML attributes for the containing element.
+ * - label_hidden: Whether to show the field label or not.
+ * - title_attributes: HTML attributes for the title.
+ * - label: The label for the field.
+ * - multiple: TRUE if a field can contain multiple items.
+ * - items: List of all the field items. Each item contains:
+ *   - attributes: List of HTML attributes for each item.
+ *   - content: The field item's content.
+ * - entity_type: The entity type to which the field belongs.
+ * - field_name: The name of the field.
+ * - field_type: The type of the field.
+ * - label_display: The display settings for the label.
+ *
+ *
+ * @see template_preprocess_field()
+ */
+#}
+{%
+  set classes = [
+    'image-gallery',
+  ]
+%}
+{%
+  set title_classes = [
+    'field__label',
+    label_display == 'visually_hidden' ? 'visually-hidden',
+  ]
+%}
+
+{% if label_hidden %}
+  {% if multiple %}
+    <ul{{ attributes.addClass(classes) }}>
+      {%- for item in items %}
+        {% block gallery %}
+          <li>{{ item.content }}</li>
+        {% endblock %}
+      {% endfor -%}
+      {% block gallery_extras %}
+          <li class="image-gallery__show-more">
+            <div class="show-more__content">
+              <span class="show-more__total">7</span>
+              <span>{{ 'Show all photos'|t }} <i class="icon-arrow-right" aria-hidden="true"></i></span>
+            </div>
+          </li>
+        </ul>
+        <span hidden="" id="image-gallery__open-gallery">{{ 'open gallery'|t }}</span>
+      {% endblock %}
+  {% else %}
+    {%- for item in items %}
+      {{ block('gallery') }}
+    {% endfor -%}
+    {{ block('gallery_extras') }}
+  {% endif %}
+{% else %}
+  <ul{{ attributes.addClass(classes) }}>
+    <div{{ title_attributes.addClass(title_classes) }}>{{ label }}</div>
```

Comment:
Only allowed children of UL are zero or more LI elements.

Reply:
Placed those divs outside the list.